Differential Gene Expression Analysis of Microarray Data
================
*Cemre Kefeli*

CEL files are raw data files generated by Affymetrix microarray
software. They contain information about the probes used in array.
Before doing a differential gene expression analysis, first, CEL files
must be processed to extract the expression values for each feature.

For this demonstration, I will use the GEO series GSE87429 as an
example.

The experiment was conducted with platform GPL17586 and has four
samples. Platforms are commercial gene chips that contain
oligonucleotides for DNA to hybridize. There are variety of different
microarray platform and it is important to know the type of platform
used in an experiment before hand in order to conduct the further
analyses.

First, we download CEL files from GEO
(<https://www.ncbi.nlm.nih.gov/geo/>) using experiment’s accession
number. The CEL files are usually in archive format (tar or zip) we have
to extract the CEL files first. Files need to be in working directory or
we can set our working directory to CEL files’ location.

To read CEL files we need Bioconductor libraries oligo, oligoClasses and
affyio.

First we’ll get the list of CEL files in directory.

``` r
cel.files <- list.celfiles()
cel.files
```

    ## [1] "GSM2331244_A_GAC-MO-A_HTA-2_0_.CEL" "GSM2331245_B_GAC-MO-B_HTA-2_0_.CEL"
    ## [3] "GSM2331246_E_GAC-MO-E_HTA-2_0_.CEL" "GSM2331247_F_GAC-MO-F_HTA-2_0_.CEL"

Then we’ll create a vector for sample names. Order of the sample name
vector must correspond to the CEL file list so that file and sample
names match correctly. It is not necessary to create sample names but
sometimes the files names can be lengthy so creating sample names may
come in handy in these situations.

``` r
sample.names <- c("control1", "treatment1", "control2", "treatment2")
```

Now we’ll read the files passing list and sample names to `read.celfile`
function and store in `affy_raw_data`.

``` r
affy_raw_data <- read.celfiles(cel.files, sampleNames = sample.names)
```

    ## Reading in : GSM2331244_A_GAC-MO-A_HTA-2_0_.CEL
    ## Reading in : GSM2331245_B_GAC-MO-B_HTA-2_0_.CEL
    ## Reading in : GSM2331246_E_GAC-MO-E_HTA-2_0_.CEL
    ## Reading in : GSM2331247_F_GAC-MO-F_HTA-2_0_.CEL

Now we’ll perform rma normalization and save the expression data file in
txt format.

``` r
rma_normalized <- rma(affy_raw_data)
```

    ## Background correcting
    ## Normalizing
    ## Calculating Expression

Now we’ll retrieve the normalized expression values for each probe.

``` r
expr <- data.frame(exprs(rma_normalized))
expr <- rownames_to_column(expr, "ProbeId")
```

<table class=" lightable-classic" style="color: black; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Annotations of the Probe Ids
</caption>
<thead>
<tr>
<th style="text-align:left;">
ProbeId
</th>
<th style="text-align:right;">
control1
</th>
<th style="text-align:right;">
treatment1
</th>
<th style="text-align:right;">
control2
</th>
<th style="text-align:right;">
treatment2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
2824546_st
</td>
<td style="text-align:right;">
12.29690
</td>
<td style="text-align:right;">
12.22003
</td>
<td style="text-align:right;">
12.36024
</td>
<td style="text-align:right;">
12.32230
</td>
</tr>
<tr>
<td style="text-align:left;">
2824549_st
</td>
<td style="text-align:right;">
11.88990
</td>
<td style="text-align:right;">
11.75083
</td>
<td style="text-align:right;">
11.87327
</td>
<td style="text-align:right;">
11.91384
</td>
</tr>
<tr>
<td style="text-align:left;">
2824551_st
</td>
<td style="text-align:right;">
12.11241
</td>
<td style="text-align:right;">
11.96338
</td>
<td style="text-align:right;">
12.11936
</td>
<td style="text-align:right;">
12.03423
</td>
</tr>
<tr>
<td style="text-align:left;">
2824554_st
</td>
<td style="text-align:right;">
11.97461
</td>
<td style="text-align:right;">
11.76430
</td>
<td style="text-align:right;">
11.85788
</td>
<td style="text-align:right;">
11.87959
</td>
</tr>
<tr>
<td style="text-align:left;">
2827992_st
</td>
<td style="text-align:right;">
11.58672
</td>
<td style="text-align:right;">
11.47740
</td>
<td style="text-align:right;">
11.54053
</td>
<td style="text-align:right;">
11.43947
</td>
</tr>
<tr>
<td style="text-align:left;">
2827995_st
</td>
<td style="text-align:right;">
11.15864
</td>
<td style="text-align:right;">
11.21040
</td>
<td style="text-align:right;">
11.25329
</td>
<td style="text-align:right;">
11.12399
</td>
</tr>
</tbody>
</table>

The expression set is loaded. We have probe names in the first column.
Now we will annotate them by finding which probe corresponding to which
gene.

#### Gene Annotations

To do the annotation first we will store column with probe ID’s in
`rma_normalized` as a vector. We will use this vector in annotation
function.

``` r
mykeys <- expr$ProbeId
head(mykeys) 
```

    ## [1] "2824546_st" "2824549_st" "2824551_st" "2824554_st" "2827992_st"
    ## [6] "2827995_st"

For annotation we will use `select` function from AnnotationDbi library.

``` r
library(AnnotationDbi)
```

Affymetrix hta20 type of microarray was used for this experiment. We
need annotation data package for this platform.

``` r
library(hta20transcriptcluster.db)
```

We can use `keytypes` function to see what kind information we can call.

``` r
keytypes(hta20transcriptcluster.db)
```

    ##  [1] "ACCNUM"       "ALIAS"        "ENSEMBL"      "ENSEMBLPROT"  "ENSEMBLTRANS"
    ##  [6] "ENTREZID"     "ENZYME"       "EVIDENCE"     "EVIDENCEALL"  "GENENAME"    
    ## [11] "GENETYPE"     "GO"           "GOALL"        "IPI"          "MAP"         
    ## [16] "OMIM"         "ONTOLOGY"     "ONTOLOGYALL"  "PATH"         "PFAM"        
    ## [21] "PMID"         "PROBEID"      "PROSITE"      "REFSEQ"       "SYMBOL"      
    ## [26] "UCSCKG"       "UNIPROT"

Since all necessary data loaded now we will do annotation and store in
`annotations`

``` r
annotations <- select(hta20transcriptcluster.db, keys = mykeys, c("SYMBOL", "ENTREZID"), "PROBEID")
annotations[sample(1:nrow(annotations), 8,replace=FALSE),] %>% 
  kbl(caption = "Annotations of the Probe Ids") %>%   
  kable_classic(full_width = F, html_font = "Cambria")
```

<table class=" lightable-classic" style="color: black; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<caption>
Annotations of the Probe Ids
</caption>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
PROBEID
</th>
<th style="text-align:left;">
SYMBOL
</th>
<th style="text-align:left;">
ENTREZID
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
17974
</td>
<td style="text-align:left;">
TC03003408.hg.1
</td>
<td style="text-align:left;">
PRSS45P
</td>
<td style="text-align:left;">
377047
</td>
</tr>
<tr>
<td style="text-align:left;">
58775
</td>
<td style="text-align:left;">
TC16001290.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
54982
</td>
<td style="text-align:left;">
TC15000380.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
65369
</td>
<td style="text-align:left;">
TC19001731.hg.1
</td>
<td style="text-align:left;">
SNAR-A4
</td>
<td style="text-align:left;">
100169956
</td>
</tr>
<tr>
<td style="text-align:left;">
2932
</td>
<td style="text-align:left;">
PSR6_qbl_hap6000473.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
43612
</td>
<td style="text-align:left;">
TC10002890.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
47755
</td>
<td style="text-align:left;">
TC12000547.hg.1
</td>
<td style="text-align:left;">
ATP23
</td>
<td style="text-align:left;">
91419
</td>
</tr>
<tr>
<td style="text-align:left;">
23090
</td>
<td style="text-align:left;">
TC05002054.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
</tr>
</tbody>
</table>

``` r
dim(annotations)
```

    ## [1] 72381     3

So we’ll drop *NA* values which are transcripts that hybridized with not
annotated probs. We will join annotation data and expression data using
dplyr function `inner_join`.

``` r
joined <- inner_join(annotations, expr, by = c("PROBEID" = "ProbeId"))
joined[sample(1:nrow(joined), 8,replace=FALSE),] %>% 
  kbl() %>%   
  kable_classic(full_width = F, html_font = "Cambria")
```

<table class=" lightable-classic" style="color: black; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
PROBEID
</th>
<th style="text-align:left;">
SYMBOL
</th>
<th style="text-align:left;">
ENTREZID
</th>
<th style="text-align:right;">
control1
</th>
<th style="text-align:right;">
treatment1
</th>
<th style="text-align:right;">
control2
</th>
<th style="text-align:right;">
treatment2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
34124
</td>
<td style="text-align:left;">
TC08002005.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
2.485613
</td>
<td style="text-align:right;">
2.219256
</td>
<td style="text-align:right;">
2.202603
</td>
<td style="text-align:right;">
2.439784
</td>
</tr>
<tr>
<td style="text-align:left;">
46202
</td>
<td style="text-align:left;">
TC11002502.hg.1
</td>
<td style="text-align:left;">
KRTAP5-AS1
</td>
<td style="text-align:left;">
338651
</td>
<td style="text-align:right;">
6.466204
</td>
<td style="text-align:right;">
6.529476
</td>
<td style="text-align:right;">
6.491683
</td>
<td style="text-align:right;">
6.417422
</td>
</tr>
<tr>
<td style="text-align:left;">
63611
</td>
<td style="text-align:left;">
TC18000997.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
6.007433
</td>
<td style="text-align:right;">
5.995465
</td>
<td style="text-align:right;">
5.665069
</td>
<td style="text-align:right;">
5.732532
</td>
</tr>
<tr>
<td style="text-align:left;">
9025
</td>
<td style="text-align:left;">
TC01005974.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
6.132620
</td>
<td style="text-align:right;">
5.596621
</td>
<td style="text-align:right;">
5.973802
</td>
<td style="text-align:right;">
5.620952
</td>
</tr>
<tr>
<td style="text-align:left;">
16405
</td>
<td style="text-align:left;">
TC03001838.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
2.970555
</td>
<td style="text-align:right;">
2.948276
</td>
<td style="text-align:right;">
2.954503
</td>
<td style="text-align:right;">
2.939036
</td>
</tr>
<tr>
<td style="text-align:left;">
1586
</td>
<td style="text-align:left;">
PSR01005530.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
12.545897
</td>
<td style="text-align:right;">
12.497638
</td>
<td style="text-align:right;">
12.515541
</td>
<td style="text-align:right;">
12.517112
</td>
</tr>
<tr>
<td style="text-align:left;">
36224
</td>
<td style="text-align:left;">
TC09001454.hg.1
</td>
<td style="text-align:left;">
ACTL7B
</td>
<td style="text-align:left;">
10880
</td>
<td style="text-align:right;">
5.285345
</td>
<td style="text-align:right;">
5.423416
</td>
<td style="text-align:right;">
5.131930
</td>
<td style="text-align:right;">
5.167648
</td>
</tr>
<tr>
<td style="text-align:left;">
16662
</td>
<td style="text-align:left;">
TC03002101.hg.1
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:right;">
4.657839
</td>
<td style="text-align:right;">
4.831276
</td>
<td style="text-align:right;">
4.423525
</td>
<td style="text-align:right;">
4.540111
</td>
</tr>
</tbody>
</table>

``` r
dim(joined)
```

    ## [1] 72381     7

Now we’ll drop the na values.

``` r
dropped <- drop_na(joined)
```

<table class=" lightable-classic" style="color: black; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
PROBEID
</th>
<th style="text-align:left;">
SYMBOL
</th>
<th style="text-align:left;">
ENTREZID
</th>
<th style="text-align:right;">
control1
</th>
<th style="text-align:right;">
treatment1
</th>
<th style="text-align:right;">
control2
</th>
<th style="text-align:right;">
treatment2
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
4706
</td>
<td style="text-align:left;">
TC02004586.hg.1
</td>
<td style="text-align:left;">
NOC2LP2
</td>
<td style="text-align:left;">
401010
</td>
<td style="text-align:right;">
6.850311
</td>
<td style="text-align:right;">
6.798528
</td>
<td style="text-align:right;">
6.941652
</td>
<td style="text-align:right;">
6.795440
</td>
</tr>
<tr>
<td style="text-align:left;">
7213
</td>
<td style="text-align:left;">
TC04001380.hg.1
</td>
<td style="text-align:left;">
FAM13A
</td>
<td style="text-align:left;">
10144
</td>
<td style="text-align:right;">
6.534445
</td>
<td style="text-align:right;">
6.860095
</td>
<td style="text-align:right;">
6.389813
</td>
<td style="text-align:right;">
6.943826
</td>
</tr>
<tr>
<td style="text-align:left;">
29662
</td>
<td style="text-align:left;">
TC20001693.hg.1
</td>
<td style="text-align:left;">
LSM14B
</td>
<td style="text-align:left;">
149986
</td>
<td style="text-align:right;">
5.532394
</td>
<td style="text-align:right;">
5.718524
</td>
<td style="text-align:right;">
5.485940
</td>
<td style="text-align:right;">
5.291141
</td>
</tr>
<tr>
<td style="text-align:left;">
20930
</td>
<td style="text-align:left;">
TC13000220.hg.1
</td>
<td style="text-align:left;">
ALG11
</td>
<td style="text-align:left;">
440138
</td>
<td style="text-align:right;">
6.576069
</td>
<td style="text-align:right;">
6.479499
</td>
<td style="text-align:right;">
6.607688
</td>
<td style="text-align:right;">
6.692807
</td>
</tr>
<tr>
<td style="text-align:left;">
6472
</td>
<td style="text-align:left;">
TC03003412.hg.1
</td>
<td style="text-align:left;">
NAA80
</td>
<td style="text-align:left;">
24142
</td>
<td style="text-align:right;">
6.734448
</td>
<td style="text-align:right;">
6.856307
</td>
<td style="text-align:right;">
6.746569
</td>
<td style="text-align:right;">
6.755015
</td>
</tr>
<tr>
<td style="text-align:left;">
14762
</td>
<td style="text-align:left;">
TC0X000079.hg.1
</td>
<td style="text-align:left;">
SYAP1
</td>
<td style="text-align:left;">
94056
</td>
<td style="text-align:right;">
8.398104
</td>
<td style="text-align:right;">
8.386778
</td>
<td style="text-align:right;">
8.418325
</td>
<td style="text-align:right;">
8.460119
</td>
</tr>
<tr>
<td style="text-align:left;">
18760
</td>
<td style="text-align:left;">
TC11002233.hg.1
</td>
<td style="text-align:left;">
MMP10
</td>
<td style="text-align:left;">
4319
</td>
<td style="text-align:right;">
3.770022
</td>
<td style="text-align:right;">
3.830922
</td>
<td style="text-align:right;">
3.737651
</td>
<td style="text-align:right;">
3.828681
</td>
</tr>
<tr>
<td style="text-align:left;">
702
</td>
<td style="text-align:left;">
TC01001161.hg.1
</td>
<td style="text-align:left;">
FCGR1CP
</td>
<td style="text-align:left;">
100132417
</td>
<td style="text-align:right;">
3.056365
</td>
<td style="text-align:right;">
3.042168
</td>
<td style="text-align:right;">
3.198485
</td>
<td style="text-align:right;">
3.090907
</td>
</tr>
</tbody>
</table>

``` r
dim(dropped)
```

    ## [1] 32124     7

There are multiple probes for the same transcript, so we have multiple
entry (observation) for the same gene. To make them unique we took the
average.

``` r
expr <- dropped[-1]
expr <- expr %>%
    group_by(SYMBOL, ENTREZID) %>% 
    summarise_each(funs(mean))
```

To see the differentially expressed genes we will use limma package. The
limma builds linear models.

We need to specify the sample groups as factor.

``` r
sample.groups <- factor(c("C", "T", "C", "T"), levels = c("C", "T"))
```

Now we’ll create design matrix.

``` r
design.mat <- model.matrix(~0 + ~sample.groups)
colnames(design.mat) = c("C", "T")
```

Using the design matrix a contrast matrix is built which will be used
while building the linear model.

``` r
contrast.matrix <- makeContrasts(C-T, levels=design.mat)
```

Using the normalized expression data, design matrix and contrast matrix
we built linear model and fit the model.

``` r
fit <- lmFit(rma_normalized, design.mat)
fit2 = contrasts.fit(fit, contrast.matrix)
fit2 = eBayes(fit2)
```

We will store the differential expression table using the `topTable`
function

``` r
table <- topTable(fit2, number = Inf, coef = 1)
head(table) %>% 
  kbl() %>%   
  kable_classic(full_width = F, html_font = "Cambria")
```

<table class=" lightable-classic" style="color: black; font-family: Cambria; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
logFC
</th>
<th style="text-align:right;">
AveExpr
</th>
<th style="text-align:right;">
t
</th>
<th style="text-align:right;">
P.Value
</th>
<th style="text-align:right;">
adj.P.Val
</th>
<th style="text-align:right;">
B
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
TC05000550.hg.1
</td>
<td style="text-align:right;">
-2.664223
</td>
<td style="text-align:right;">
7.489702
</td>
<td style="text-align:right;">
-33.70718
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0008736
</td>
<td style="text-align:right;">
8.382083
</td>
</tr>
<tr>
<td style="text-align:left;">
TC05002579.hg.1
</td>
<td style="text-align:right;">
-2.659631
</td>
<td style="text-align:right;">
8.554089
</td>
<td style="text-align:right;">
-32.97093
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0008736
</td>
<td style="text-align:right;">
8.324418
</td>
</tr>
<tr>
<td style="text-align:left;">
TC06000135.hg.1
</td>
<td style="text-align:right;">
2.414019
</td>
<td style="text-align:right;">
8.405235
</td>
<td style="text-align:right;">
29.93135
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:right;">
0.0010717
</td>
<td style="text-align:right;">
8.052560
</td>
</tr>
<tr>
<td style="text-align:left;">
TC04000478.hg.1
</td>
<td style="text-align:right;">
-2.096694
</td>
<td style="text-align:right;">
5.713626
</td>
<td style="text-align:right;">
-27.82441
</td>
<td style="text-align:right;">
1e-07
</td>
<td style="text-align:right;">
0.0012729
</td>
<td style="text-align:right;">
7.826071
</td>
</tr>
<tr>
<td style="text-align:left;">
TC07000810.hg.1
</td>
<td style="text-align:right;">
-2.232153
</td>
<td style="text-align:right;">
7.622238
</td>
<td style="text-align:right;">
-26.22526
</td>
<td style="text-align:right;">
1e-07
</td>
<td style="text-align:right;">
0.0013466
</td>
<td style="text-align:right;">
7.628687
</td>
</tr>
<tr>
<td style="text-align:left;">
TC15002249.hg.1
</td>
<td style="text-align:right;">
-1.929155
</td>
<td style="text-align:right;">
6.206933
</td>
<td style="text-align:right;">
-25.85557
</td>
<td style="text-align:right;">
1e-07
</td>
<td style="text-align:right;">
0.0013466
</td>
<td style="text-align:right;">
7.579506
</td>
</tr>
</tbody>
</table>
